### PR TITLE
feat(provider): [NV-2961] Add Google Chat Chat Provider #4381

### LINKED ISSUE #4381: [NV-2961] 🚀 Feature: Add Google Chat Chat Provider
### 🔖 Feature description

Use the [Google Chat docs](https://developers.google.com/chat/api/guides/v1/messages/create) and the [SDK](https://github.com/google/google-api-javascript-client) to create a new chat provider while following [Novu's guide to a new provider](https://github.com/novuhq/novu/issues/new?assignees=&labels=feature&projects=&template=feature_request.yml&title=%F0%9F%9A%80+Feature%3A+)

### 🎤 Why is this feature needed ?

provide more options to the community

### ✌️ How do you aim to achieve this?

community Contribution

### 🔄️ Additional Information

_No response_

### 👀 Have you spent some time to check if this feature request has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Contributing Guidelines](https://github.com/novuhq/novu/blob/main/CONTRIBUTING.md)

### Are you willing to submit PR?

None

<sub>[NV-2961](https://linear.app/novu/issue/NV-2961/🚀-feature-add-google-chat-chat-provider)</sub>

### PR DESCRIPTION
**Pull request by Sushmitha-93**
### What change does this PR introduce?

Adds the Google Chat Provider.

Referred the following Google chat webhook documentation
[Google Chat Webhook](https://developers.google.com/chat/how-tos/webhooks)

### Why was this change needed?

Closes #4381 

### Other information (Screenshots)

Integration-stores

Google chat workflow

build-successful

### CODE REVIEW

File: providers/google-chat/README.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Novu GoogleChat Provider
+
+A GoogleChat chat provider library for [@novu/node](https://github.com/novuhq/novu)
+
+## Usage
+
+```javascript
+import { GoogleChatProvider } from '@novu/google-chat';
+
+const provider = new GoogleChatProvider({});
```
Comment (Cliftonz) [inline providers/google-chat/README.md:10]: The angle brackets should not be here, should just be ().

**Review by Cliftonz — CHANGES_REQUESTED** [review]
Just one change looks great otherwise! Good work!

File: providers/google-chat/README.md (hunk 1 above)
Comment (Sushmitha-93) [reply providers/google-chat/README.md]: Thanks @Cliftonz, I updated it.

**Review by Cliftonz — APPROVED** [review]

**Comment by davidsoderberg** [comment]
Could you please provide a screenshot of how it looks in multi integration list set line 34 to true in useFeatureFlags file?

**Comment by Sushmitha-93** [comment]
> Could you please provide a screenshot of how it looks in multi integration list set line 34 to true in useFeatureFlags file?

Hi @davidsoderberg , this is how it was looking after I set line 34 to true in useFeatureFlags file. 

Integration-store-with-useFeatureFlag-line34-true


I created a new PR #4381 and closed this one because I ended up starting with a new fork.

**Pull request closed without merging**